src/web-components/thumbnail-panel.tsx

```diff
@@ -0,0 +1,46 @@
+import register from '../lib/preact-custom-element/preact-custom-element';
+import React, { useLayoutEffect, useRef } from 'react';
+import { ThumbnailPanel } from '../components/ThumbnailPanel';
+import { useCustomEvent } from './helpers/use-custom-event';
+
+interface ThumbnailPanelAttributes {
+  'iiif-content': string;
+  'current-resource-id'?: string;
+}
+
+interface ThumbnailPanelProps {
+  iiifContent: string;
+  currentResourceId?: string;
+  __registerPublicApi: (component: any) => void;
+}
+
+function ThumbnailPanelWebComponent(props: ThumbnailPanelProps & ThumbnailPanelAttributes) {
+  const webComponent = useRef<HTMLElement>();
+
+  useLayoutEffect(() => {
+    if (props.__registerPublicApi) {
+      props.__registerPublicApi((component: any) => {
+        webComponent.current = component;
+      });
+    }
+  }, []);
+
+  const handleOnChange = useCustomEvent(webComponent, 'resource-changed', (e) => ({ resourceId: e }));
+
+  return <ThumbnailPanel {...props} onResourceChanged={handleOnChange} orientation={'vertical'} />;
+}
+
+const thumbnailPanelProps = ['iiif-content', 'current-resource-id'];
```

Comment:
This is a required list of "reactive" props on the component. If it's not in this list, then it will not be observed and update in the WC.